api/envoy/config/route/v3/route_components.proto

```diff
@@ -388,6 +388,22 @@ message RouteMatch {
     google.protobuf.BoolValue validated = 2;
   }
 
+  // [#not-implemented-hide:]
+  // Configuration for sending data upstream as a raw data payload. This is used for
+  // CONNECT requests, when forwarding CONNECT payload as raw TCP.
+  message ProxyConfig {
+    // TODO(alyssawilk) add proxy proto configuration here.
+  }
+
+  // [#not-implemented-hide:]
+  // Configuration for proxying CONNECT requests.
+  message ConnectConfig {
```

Comment:
I'm concerned that we are conflating match/action here. Elsewhere in route config, we have a clear delineation of match and action. Can we split this such that we have a connect match config, which just says "match CONFIG" and then a `ConnectAction` or something like that in the `oneof action` above? 

I realize this is less direct, but otherwise this is confusing to folks trying trying to understand the basic match-action model to have an exception like this.